### ISSUE TITLE
DateFormat doesn't work with scriptCode

### ISSUE THREAD

**Issue by greymag** [flutter/flutter#81905]
## Steps to Reproduce

1. Run `flutter create date_format_bug`.
2. Update the files as follows:

**`pubspec.yaml`**
- Add dependency:
```yaml
  flutter_localizations:
    sdk: flutter
``` 

**`lib/main.dart`:**
- Add imports:
```dart
import 'package:intl/intl.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
```

- Change `MyApp` as follows:
```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final locale = Locale.fromSubtags(languageCode: 'sr', scriptCode: 'Latn');
    Intl.defaultLocale = locale.toString();
    return MaterialApp(
      title: 'Flutter Demo',
      supportedLocales: [locale],
      locale: locale,
      localizationsDelegates: [
        GlobalMaterialLocalizations.delegate,
      ],
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}
```

- Change `_MyHomePageState` as follows:
```dart
class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    final text = DateFormat.yMMMMEEEEd().format(DateTime(2021, 1, 2));
    return Scaffold(
      body: Center(
        child: Text(text),
      ),
    );
  }
}
```

3. Run the app.

**Expected results:** "subota, 2. januar 2021." on screen (in Latin).

**Actual results:** "субота, 2. јануар 2021." on screen (in Cyrillic).

## Problem

The problem is in `flutter_localizations` package, file `src/utils/date_localizations.dart`, function `loadDateIntlDataIfNotLoaded()`, where `scriptCode` simply has been removed, so `sr_Latn` data is just skipped (because `sr` already exists).

```dart
void loadDateIntlDataIfNotLoaded() {
       ...
      .forEach((String locale, Map<String, dynamic> data) {
        // Strip scriptCode from the locale, as we do not distinguish between scripts
        // for dates.
        final List<String> codes = locale.split('_');
        String? countryCode;
        if (codes.length == 2) {
          countryCode = codes[1].length < 4 ? codes[1] : null;
        } else if (codes.length == 3) {
          countryCode = codes[1].length < codes[2].length ? codes[1] : codes[2];
        }
        locale = codes[0] + (countryCode != null ? '_' + countryCode : '');
        if (initializedLocales.contains(locale))
          return;
        ...
      });
    ...
  }
}
```

There are TODO in there, but problem still exists and brokes some localizations.

<details>
  <summary>Logs</summary>

`flutter analyze`:
```
Analyzing date_format_bug...                                            
No issues found! (ran in 4.2s)
```

`flutter doctor -v`:
```
[✓] Flutter (Channel stable, 2.0.5, on Mac OS X 10.15.7 19H524 darwin-x64, locale en-RU)
    • Flutter version 2.0.5 at /Volumes/MacbookExt/Development/sdk/flutter_2.0.1
    • Framework revision adc687823a (3 weeks ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/GreyMag/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/GreyMag/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/GreyMag/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Ultimate Edition (version 2017.3.5)
    • IntelliJ at /Users/GreyMag/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.55.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • EFDA5575-9127-47A4-B5CE-F6D0F9E716B2 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 90.0.4430.93

• No issues found!
```

</details>

**Comment by TahaTesser** [comment]
Hi @greymag 
Thanks for filing the issue, I can confirm the issue.

Here I assembled the pieces of your code into a runnable code sample

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import 'package:flutter_localizations/flutter_localizations.dart';
void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final locale = Locale.fromSubtags(languageCode: 'sr', scriptCode: 'Latn');
    Intl.defaultLocale = locale.toString();
    return MaterialApp(
      title: 'Flutter Demo',
      supportedLocales: [locale],
      locale: locale,
      localizationsDelegates: [
        GlobalMaterialLocalizations.delegate,
      ],
      home:Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  @override
  Widget build(BuildContext context) {
    final text = DateFormat.yMMMMEEEEd().format(DateTime(2021, 1, 2));
    return Scaffold(
      body: Center(
        child: Text(text),
      ),
    );
  }
}
``` 

</details>


<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.3 20E232 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 1d9032c7e1 (6 days ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 4D5BE4C8-BF1D-46CD-B595-FD81A7140B52 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • macOS 11.3 20E232 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 90.0.4430.93

• No issues found!
```

```console
[✓] Flutter (Channel master, 2.3.0-1.0.pre.227, on macOS 11.3 20E232 darwin-x64,
    locale en-GB)
    • Flutter version 2.3.0-1.0.pre.227 at /Users/tahatesser/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7f1d1414cc (26 hours ago), 2021-05-04 06:17:19 -0700
    • Engine revision 378e4dbc41
    • Dart version 2.14.0 (build 2.14.0-48.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2021.1.1)
    • IntelliJ at /Volumes/Extreme/IntelliJ IDEA CE.app
    • Flutter plugin version 55.1.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • 4D5BE4C8-BF1D-46CD-B595-FD81A7140B52 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • macOS 11.3 20E232 darwin-x64
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 90.0.4430.93

• No issues found!

```
		
</details>

**Comment by HansMuller** [comment]
CC @shihaohong

**Comment by greymag** [comment]
It was fixed in PR #83903 and merged in master a long time ago. Stable Flutter doesn't have this problem anymore.
